### ISSUE TITLE
Extra calls to API Server in some scenarios. Ideally, we'd get all pods in the namespace only once.

### ISSUE THREAD

**Issue by gabrielggg** [kubernetes-sigs/network-policy-api#236]
Just noting that there could be lots of extra calls to API Server in some scenarios. Ideally, we'd get all pods in the namespace only once.

Could you file a GitHub issue to optimize this in another PR?

_Originally posted by @huntergregory in https://github.com/kubernetes-sigs/network-policy-api/pull/227#discussion_r1645048124_
